config.py

```diff
@@ -95,6 +96,8 @@ def __init__(self) -> None:
         with open(f"{folder_path}/config.json", "w", encoding="utf-8-sig") as file:
             json.dump(self.config, file, indent=4)
 
+        self.__resolve_automatic_settings()
```

Comment:
note: this is happening _after_ we dump the loaded config into the result directory as JSON, meaning that will still have the `auto` properties

I think that is probably better than doing the resolving before that because we do our printing of the config values into the audit log after this (so we already have the resolved values somewhere), and "auto" is meant to be reliable so I think keeping the "auto" is nicer for a maintainer trying to triage a bug report by running with someones submitted config

(and if "auto" is ever _not_ reliable, then that itself is a problem to be addressed)

Reply:
Sounds cleaner